packages/@aws-cdk/aws-stepfunctions-tasks/lib/dynamodb/shared-types.ts

```diff
@@ -232,6 +232,14 @@ export class DynamoAttributeValue {
     return new DynamoAttributeValue({ BOOL: value });
   }
 
+  /**
+   * maps a boolean from a string - can be a literal or a json path
```

Comment:
I think this should not support string literals and validate the JsonPath similar to how `mapFromJsonPath` method is laid out.

If it turns out we really need it, we can always expand later. I can't really think of a scenario where a string literal would be needed as all supported languages should be able to use `fromBoolean` instead. It also helps prevent user errors from providing strings like "foo" when only "true" and "false" are valid.

You could extract that validation logic from `mapFromJsonPath` into the `.private` folder

Reply:
I agree - makes a lot of sense

Reply:
Are you saying it should actually take a JsonPath? or is confirming it starts with $ adequate in this regard?

Reply:
yep, validating that it starts with `$` is what I meant which is roughly the same as accepting only Json paths